content/getting-work-done-in-open-source.rst

```diff
@@ -0,0 +1,157 @@
+in the shoes of the submitter, who you most likely have never met, whose skill
+level and experience with the code base you cannot judge, and try to understand
+their problem and see the solution from their eyes. If the change is
+particularly hairy and the submitter has not explained it very well, it's
+tempting to gloss over the change and pick on the trivial things, or worse to
+just ignore the change.
+
+While the project maintainers should carry most of the responsibility for
+properly guiding a new contributor through the project's contribution processes
+and helping them to meet the maintainers expectations, there is a foolproof way
+that code submitters, regardless of the project, can speed the process along:
+
+**Write better commit messages**
+
+The most important part of a code change is not the code itself but the commit
+message. The prose you use to prepare the reviewer for the change is your best
+opportunity to get them on your side. Perfect spelling and conformity to style
+conventions are not what is important here (but I do still recommend you read up
+on the `guidelines`_). What matters is getting the discussion off on the right
+foot. The following is what has worked for me as a code contributor to get my
+work reviewed and merged faster, and what helps me as a reviewer to more quickly
+digest and respond to change proposals.
+
+.. _guidelines: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
+
+How to write good commit messages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A good commit message comes in four parts::
+
+  What changed
+
+  "Without this patch..."
+
+  "This is a problem because..."
+
+  "This patch solves the problem by..."
+
+The subject of the commit message is a brief but clear description of what
+changed. It is not enough to say "Fix bug #123". The subject must be descriptive
+enough to stand on its own, because in some cases it may be the only resource
+available to explain what changed. For instance, the subject line is often the
+only part of a change that is pulled into a changelog, whether it's your own
+changelog or a downstream package's changelog. It's also extremely useful for
+the subject to be descriptive enough that you can look through a git history and
+pinpoint a moment in time just by skimming the subject lines. A subject line
+that identifies the component that was changed and how is going to be much more
+helpful when it comes time to debug a test failure in that component than will a
+generic line about fixing a bug.
+
+The body of the commit message has at least three parts. The first part
+describes the state of the world as it is today, without the change applied.
+This is where you explain the bug you noticed and the symptons you saw, perhaps
+even how to reproduce it. Or discuss how the project is lacking an important
+ability. This sets the reviewer up with context; if nothing else in the commit
```

Comment:
Maybe "feature" is more natural than "ability" here? Just an idea, feel free to ignore.